### PR TITLE
TEXstudio, new recipe

### PR DESCRIPTION
**Pull request by Begasus**
Worked ok for me on _x86, first draft was done by SirMik

### CODE REVIEW

File: app-office/texstudio/patches/texstudio-2.12.14.patchset [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/texstudio.pro b/texstudio.pro
+index da5de73..3f1bbd8 100644
+--- a/texstudio.pro
++++ b/texstudio.pro
+@@ -429,7 +429,7 @@ exists(./.git)  {
+ #QMAKE_CXXFLAGS_DEBUG += -Werror  -Wall -Wextra -Winit-self -Wmain -Wmissing-include-dirs -Wtrigraphs -Wunused -Wunknown-pragmas -Wundef -Wpointer-arith -Wtype-limits -Wwrite-strings -Wclobbered -Wempty-body -Wsign-compare -Waddress -Wlogical-op -Winline
+ QMAKE_CXXFLAGS_DEBUG += -Wall -Wextra -Winit-self -Wmissing-include-dirs -Wtrigraphs -Wunused -Wunknown-pragmas -Wundef -Wpointer-arith -Wwrite-strings -Wempty-body -Wsign-compare -Waddress -Winline
+ QMAKE_CXXFLAGS += -std=c++0x
+-!win32: QMAKE_LFLAGS += -rdynamic # option not supported by mingw
++!win32: QMAKE_LFLAGS += # option not supported by mingw
```
Comment (diversys) [inline app-office/texstudio/patches/texstudio-2.12.14.patchset]: I think you ca get away with `!win32:!haiku` here.

Comment (Begasus) [reply app-office/texstudio/patches/texstudio-2.12.14.patchset]: Didn't check that back then :)

File: app-office/texstudio/texstudio-2.12.16.recipe [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	texstudio$secondaryArchSuffix = $portVersion
+	cmd:texstudio = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+#	lib:libpoppler$secondaryArchSuffix
```
Comment (korli) [inline app-office/texstudio/texstudio-2.12.16.recipe]: this can be removed.

File: app-office/texstudio/texstudio-2.12.16.recipe [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libQt5Xml$secondaryArchSuffix
+	lib:libX11$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libGL$secondaryArchSuffix
+#	devel:libpoppler$secondaryArchSuffix
```
Comment (korli) [inline app-office/texstudio/texstudio-2.12.16.recipe]: this can be removed.

File: app-office/texstudio/texstudio-2.12.16.recipe (hunk 2 above)
Comment (Begasus) [reply app-office/texstudio/texstudio-2.12.16.recipe]: Hence it was disabled for checking :) (as the next one)
We could use system hunspell and quazip, but it's hard to check here due to the project size/compilation time.

**Comment by Begasus** [comment]
Previous version seemed to build ok here, this is the latest version but I'm getting an error on QObject on this one, used system hunspell and quazip in the recipe (not updated here): https://pastebin.com/hyuE8Fa7

**Comment by extrowerk** [comment]
64 bit:
```
In file included from /packages/qt5-5.13.1-1/.self/develop/headers/QtGui/qsortfilterproxymodel.h:45,
                 from /packages/qt5-5.13.1-1/.self/develop/headers/QtGui/QSortFilterProxyModel:1,
                 from src/symbolpanel/proxymodels.h:4,
                 from src/symbolpanel/proxymodels.cpp:1:
/boot/system/develop/headers/QtGui/qabstractproxymodel.h:58:40: error: expected initializer before ':' token
 class Q_GUI_EXPORT QAbstractProxyModel : public QAbstractItemModel
                                        ^
In file included from /boot/system/develop/headers/QtCore/qglobal.h:105,
                 from /boot/system/develop/headers/QtCore/qatomic.h:41,
                 from /boot/system/develop/headers/QtCore/qvariant.h:43,
                 from /boot/system/develop/headers/QtCore/qabstractitemmodel.h:43,
                 from /boot/system/develop/headers/QtGui/qabstractproxymodel.h:45,
                 from /packages/qt5-5.13.1-1/.self/develop/headers/QtGui/qsortfilterproxymodel.h:45,
                 from /packages/qt5-5.13.1-1/.self/develop/headers/QtGui/QSortFilterProxyModel:1,
                 from src/symbolpanel/proxymodels.h:4,
                 from src/symbolpanel/proxymodels.cpp:1:
/boot/system/develop/headers/QtGui/qabstractproxymodel.h:60:5: error: expected declaration before end of line
     Q_OBJECT
     ^~~~~~~~
Makefile:98344: recipe for target '.obj/proxymodels.o' failed
```

**Comment by Begasus** [comment]
Checked a local build on x86_64 today, worked fine now?

**Comment by Begasus** [comment]
Think this should do it, builds ok on gcc2h (secondaryArchitecture) and x86_64 now :)

**Comment by korli** [comment]
Why is it disabled?

**Comment by Begasus** [comment]
> Why is it disabled?

whoops, need to change (disabled it because of the previous error which seems to be fixed) :)

**Comment by Begasus** [comment]
Build texlive here (call me crazy) :) but it's working fine it seems :+1: 
Te-Xstudio-La-Te-X2

**Pull request merged**